flang/test/Lower/character-assignment.f90

```diff
@@ -1,44 +1,47 @@
-! RUN: bbc %s -o "-" -emit-fir | FileCheck %s
+! RUN: bbc %s -o - -emit-fir | FileCheck %s
+! RUN: bbc %s -o - | FileCheck --check-prefix=UNBOX %s
```

Comment:
Great question. This was a headache, in fact. Because we are passing around CHARs as boxchar values, and we have the new symbol map code, Burnside is lowering CHAR dummies in two parallel ways at the same time (for now). This broke this test because it means we have two unboxchar Ops for each CHAR dummy. The second pass allows the cleanup passes to run, so it gets rid of the unused unboxchar and the UNBOX checks match correctly. Other parts of this test are matching for the high-level constructs, so running the cleanups would have washed those away causing those CHECKs to fail.

This is not a perfect solution. We need to talk more about threading results through Burnside.

Reply:
Mmmh, OK. I agree a single mlir::Value will not suffice anymore.